src/main/scala/com/horizon/exchangeapi/CatalogRoutes.scala

```diff
@@ -250,6 +275,314 @@ trait CatalogRoutes extends JacksonSupport with AuthenticationSupport {
+        exchAuth(TNode(OrgAndId(orgid,"#").toString), Access.READ) {
+          ident =>
+            validateWithMsg(GetNodesUtils.getNodesProblem(nodeType)) {
+              complete({
+                implicit val jsonFormats: Formats = DefaultFormats
+                val ownerFilter: Option[String] =
+                  if(ident.isAdmin || ident.role.equals(AuthRoles.Agbot))
+                    owner
+                  else
+                    Some(ident.identityString)
+      
+                val getNodes =
+                  for {
+                    nodes ← NodesTQ.rows
+                                   .filterOpt(arch)((node, arch) ⇒ node.arch like arch)
+                                   .filterOpt(id)((node, id) ⇒ node.id like id)
+                                   .filterOpt(name)((node, name) ⇒ node.name like name)
+                                   .filterOpt(nodeType)(
+                                     (node, nodeType) ⇒ {
+                                       (node.nodeType === nodeType.toLowerCase ||
+                                        node.nodeType === nodeType.toLowerCase.replace("device", ""))})
+                                   .filter(_.orgid === orgid)
+                                   .filterOpt(ownerFilter)((node, ownerFilter) ⇒ node.owner like ownerFilter)
+                                   .joinLeft(NodeErrorTQ.rows.filterOpt(id)((nodeErrors, id) ⇒ nodeErrors.nodeId like id))
+                                     .on(_.id === _.nodeId)
+                                   .joinLeft(NodeStatusTQ.rows.filterOpt(id)((nodeStatuses, id) ⇒ nodeStatuses.nodeId like id))
+                                     .on(_._1.id === _.nodeId)
+                                   .sortBy(_._1._1.id.asc)
+                                   .map(
+                                     node ⇒
+                                       (node._1._1.arch,
+                                        node._1._1.id,
+                                        node._1._1.heartbeatIntervals,
+                                        node._1._1.lastHeartbeat,
+                                        node._1._1.lastUpdated,
+                                        node._1._1.msgEndPoint,
+                                        node._1._1.name,
+                                        node._1._1.nodeType,
+                                        node._1._1.orgid,
+                                        node._1._1.owner,
+                                        node._1._1.pattern,
+                                        node._1._1.publicKey,
+                                        node._1._1.regServices,
+                                        node._1._1.softwareVersions,
+                                        (if(ident.isSuperUser)
+                                          node._1._1.id.substring(0,0)
```

Comment:
Add a comment here about how you're not even pulling the node token from the db

Reply:
Maybe also add a check here for if the user is an org admin, org admins can also see all nodes in an org